src/action/ActivateKeyValueAuthentication.php

```diff
@@ -123,6 +124,7 @@ private function registerAuthKeyValueAdapter()
     private function registerUserEventListener(): void
     {
         $this->registerEvent(UserUpdatedEvent::class, [UserEventListener::class, 'userUpdated']);
+        $this->registerEvent(ResourceUpdated::class, [UserEventListener::class, 'userUpdated']);
```

Comment:
why do not trigger `UserUpdatedEvent` when update test center instead of additionally handling more generic `ResourceUpdated` here?

Reply:
It related to generic tree logic and how we save resources which were updated from a tree 
https://github.com/oat-sa/tao-core/blob/master/actions/class.GenerisTree.php#L70
As we have only simple update for all properties for any arrived resource I'm not sure that we need to add that dependency on the core side, or move all that logic to a service and reuse it on the auth key value side 
https://github.com/oat-sa/tao-core/blob/master/actions/class.GenerisTree.php#L95.